semver/operators/lt_test.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { assertEquals } from "../../testing/asserts.ts";
+import { parse } from "../parse.ts";
+import { lt } from "./lt.ts";
+
+Deno.test({
+  name: "comparison",
+  fn: async (t) => {
+    // [version1, version2]
+    // version1 should be greater than version2
+    const versions: [string, string][] = [
+      ["0.0.0", "0.0.0-foo"],
+      ["0.0.1", "0.0.0"],
+      ["1.0.0", "0.9.9"],
+      ["0.10.0", "0.9.0"],
+      ["0.99.0", "0.10.0"],
+      ["2.0.0", "1.2.3"],
+      ["1.2.3", "1.2.3-asdf"],
+      ["1.2.3", "1.2.3-4"],
+      ["1.2.3", "1.2.3-4-foo"],
+      ["1.2.3-5", "1.2.3-5-foo"], // numbers > strings, `5-foo` is a string not a number
+      ["1.2.3-5", "1.2.3-4"],
+      ["1.2.3-5-foo", "1.2.3-5-Foo"],
+      ["3.0.0", "2.7.2+asdf"],
+      ["1.2.3-a.10", "1.2.3-a.5"],
+      ["1.2.3-a.5", "1.2.3-a.b"],
```

Comment:
Interesting, good catch.

Here is the place in code actually doing this comparison:
https://github.com/justinmchase/deno_std/blob/0421330f380d21486b239d6b813b2e6344599bb4/semver/operators/compare.ts#L73

```ts
    } else if (typeof a === "string" && typeof b === "number") {
      // string < number
      return -1;
    } else if (typeof a === "number" && typeof b === "string") {
      // number > string
      return 1;
    } else if (a < b) {
```

Which is indeed the _opposite_ of what the spec is saying. If I recall correctly and look in the original code, this was actually an untested situation and it was simply doing `a < b` and it was actually a bug because...

```ts
"a" < 1 // false
1 < "a" // false
```

Which then I either misinterpretted or inverted the spec somewhere along the way.